lib/krb5/get_cred.c

```diff
@@ -1051,9 +1051,7 @@ get_cred_kdc_referral(krb5_context context,
 	}
 
 	/* Did we get the right ticket ? */
-	if (krb5_principal_compare_any_realm(context,
-					     referral.server,
-					     ticket.server))
+	if (krb5_principal_compare(context, referral.server, ticket.server))
```

Comment:
FYI, MIT does allow completely different ticket sname@srealm when canonicalization is requested, but IIRC this requires FAST and other functionality we don't have in Heimdal.